src/settings/NumberInput.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+type NumberInputProps = {
+  label: string;
+  value: number;
+  onChange: (x: number) => void;
+};
+
+export default function NumberInput({
+  label,
+  value,
+  onChange,
+}: NumberInputProps) {
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    if (Number.isSafeInteger(Number(value))) {
+      onChange(Number(value));
+    }
+  };
+
+  return (
+    <React.Fragment>
+      <input
+        type='text'
```

Comment:
Why is NumberInput's input's type `'text'`? You should also add a `max` attribute to indicate that there is a maximum

Reply:
I think it's more  easy to style normal input that stylling number input, but yeah you're absolutely right.